File: library/src/main/java/com/mikepenz/storyblok/model/Tag.java
package com.mikepenz.storyblok.model;

import org.json.JSONArray;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by mikepenz on 14/04/2017.
 */

public class Tag extends Entity {
    private int taggingsCount;

    public Tag(JSONObject story) {
        super(story);

        if (story.has("taggings_count")) {
            this.taggingsCount = story.optInt("taggings_count");
        }
    }

    public static List<Tag> parseTags(JSONObject result) {
        if (result != null && result.has("tags")) {
            JSONArray jsonArray = result.optJSONArray("tags");
            List<Tag> tags = new ArrayList<>(jsonArray.length());
            for (int i = 0; i < jsonArray.length(); i++) {
                tags.add(new Tag(jsonArray.optJSONObject(i)));
            }
            return tags;
        }
        return null;
    }

    public int getTaggingsCount() {
        return taggingsCount;
    }
}

File: library/src/main/java/com/mikepenz/storyblok/model/Tag.kt
package com.mikepenz.storyblok.model

import org.json.JSONObject
import java.util.*

/**
 * Created by mikepenz on 14/04/2017.
 */

class Tag(story: JSONObject) : Entity(story) {
    var taggingsCount: Int = 0
        private set

    init {

        if (story.has("taggings_count")) {
            this.taggingsCount = story.optInt("taggings_count")
        }
    }

    companion object {

        fun parseTags(result: JSONObject?): List<Tag>? {
            if (result != null && result.has("tags")) {
                val jsonArray = result.optJSONArray("tags")
                val tags = ArrayList<Tag>(jsonArray.length())
                for (i in 0 until jsonArray.length()) {
                    tags.add(Tag(jsonArray.optJSONObject(i)))
                }
                return tags
            }
            return null
        }
    }
}
